### PR TITLE
change unnecessarily longer array

### PR DESCRIPTION
**Pull request by rahul003**
simplifies code by removing the possibility of unused variables. This confused g++ earlier as some variables were uninitialized and unused. Removes three compiler warnings.

### CODE REVIEW

File: src/io/iter_image_recordio_2.cc [hunk 1]
```diff
@@ -363,7 +363,7 @@ inline void ImageRecordIOParser2<DType>::ParseChunk(dmlc::InputSplit::Blob * chu
       for (int i = 0; i < res.rows; ++i) {
         uchar* im_data = res.ptr<uchar>(i);
         for (int j = 0; j < res.cols; ++j) {
-          DType RGBA[4];
+          DType* RGBA = new DType[n_channels];
```
Comment (szha) [inline src/io/iter_image_recordio_2.cc]: This seems to be allocating new space for each pixel. Would it be possible to alloc once and reuse the RGBA buffer for all pixels?

Comment (rahul003) [inline src/io/iter_image_recordio_2.cc]: Fixed it

**Comment by piiswrong** [comment]
Why use heap when stack is enough?

**Comment by rahul003** [comment]
I initialized like this because variable length array declaration like DType[n_channels] are not technically allowed by C++ standard. I read that g++ does allow it as an extension (if we don't use the flag ```-pedantic```. I have verified that DType[n_channels] does work on g++4.8 and g++5 on Linux. I wasn't sure if we can rely on this flag not being used. 

The alternative to the above would be dynamically create array using new.

Should I just declare it as DTypes[n_channels]. It does work with the default config flags and config.mk configs.

**Comment by szha** [comment]
@rahul003 just keep the size as a constant 4 should be fine. It's constant space, and n_channels is either 1, 3 or 4. Saving at most 3*sizeof(DType) doesn't change much, and in the alternative, you can keep the array on stack.

**Comment by rahul003** [comment]
Space wise it should be fine yes, but the goal was to address compiler warnings like below.
What do you think the best way to do remove these warnings would be?

```
src/io/iter_image_recordio_2.cc:407:17: warning: 'RGBA[1]' may be used uninitialized in this function [-Wmaybe-uninitialized]
src/io/iter_image_recordio_2.cc:407:17: warning: 'RGBA[3]' may be used uninitialized in this function [-Wmaybe-uninitialized]
src/io/iter_image_recordio_2.cc:407:17: warning: 'RGBA[2]' may be used uninitialized in this function [-Wmaybe-uninitialized]
```

**Comment by szha** [comment]
`DType RGBA[4] = {}` or use `std::fill_n` after declaration should handle the initialization.

File: src/io/iter_image_recordio_2.cc [hunk 2]
```diff
@@ -360,10 +360,10 @@ inline void ImageRecordIOParser2<DType>::ParseChunk(dmlc::InputSplit::Blob * chu
           (rand_uniform(*(prnds_[tid])) * normalize_param_.max_random_illumination * 2
           - normalize_param_.max_random_illumination) * normalize_param_.scale;
       }
+      DType RGBA[4] = {}
```
Comment (szha) [inline src/io/iter_image_recordio_2.cc]: semicolon

Comment (rahul003) [inline src/io/iter_image_recordio_2.cc]: Oops, sorry. I tested it in my machine, but updated branch through website I didn't notice.

**Comment by rahul003** [comment]
Created new PR with changes at https://github.com/apache/incubator-mxnet/pull/7294

**Pull request closed without merging**